src/map/utils/trustutils.cpp

```diff
@@ -693,6 +693,15 @@ namespace trustutils
             }
         }
 
+        if (mJob == JOB_PLD)
```

Comment:
No good. we have several pld trust that do not block (no shield in model). this isn't going to fly unless you prove to me the spear wielding paladins and Monberaux (also a pld!) are secretly blocking invisibly.

Reply:
That is a very valid and completely overlooked complication.

Reply:
Suggestion: instead of setting their blockrate statically to 45, use a modifier (and can throw the 45 to it by default), then check that the modifier is non-zero here.

There may already exist one intended for a bonus on players that might be usable this way just on trusts.

Reply:
Mob:setMod(xi.mod.SHIELDBLOCKRATE) is an existing mod we could use.

Reply:
and only used in 2 spots both in battleutils. one in GetBlockRate(0 and one in a check specifically for pets (automaton). the auto has a similar problem: its shield is fake. So this seems perfect.

Reply:
Ok, I'll test using the mod instead this evening and adjust the PR based on results.

Reply:
what I messed with yesterday. applies block, applies spikes. open to ideas / changes

is blocked, adjusted whole block to bypass the gear check and then fall through reprisal

```
if (isBlocked)
            {
                int16 blockRateMod = PDefender->getMod(Mod::SHIELDBLOCKRATE);
                uint8 absorb = 100;
                if (PDefender->m_Weapons[SLOT_SUB]->IsShield())
                {
                    if (PDefender->objtype == TYPE_PC)
                    {
                        absorb = std::clamp(100 - PDefender->m_Weapons[SLOT_SUB]->getShieldAbsorption(), 0, 100);
                        absorb -= PDefender->getMod(Mod::SHIELD_DEF_BONUS); // Include Shield Defense Bonus in absorb amount

                        // Shield Mastery
                        if ((std::max(damage - (PDefender->getMod(Mod::PHALANX) + PDefender->getMod(Mod::STONESKIN)), 0) > 0) &&
                            charutils::hasTrait((CCharEntity*)PDefender, TRAIT_SHIELD_MASTERY))
                        {
                            // If the player blocked with a shield and has shield mastery, add shield mastery TP bonus
                            // unblocked damage (before block but as if affected by stoneskin/phalanx) must be greater than zero
                            PDefender->addTP(PDefender->getMod(Mod::SHIELD_MASTERY_TP));
                        }
                    }
                }
                else if (PDefender->objtype == TYPE_PET && blockRateMod > 0)
                {
                    absorb = 50;

                    // Shield Mastery
                    if ((std::max(damage - (PDefender->getMod(Mod::PHALANX) + PDefender->getMod(Mod::STONESKIN)), 0) > 0) &&
                        (PDefender->getMod(Mod::SHIELD_MASTERY_TP)))
                    {
                        // If the player blocked with a shield and has shield mastery, add shield mastery TP bonus
                        // unblocked damage (before block but as if affected by stoneskin/phalanx) must be greater than zero
                        PDefender->addTP(PDefender->getMod(Mod::SHIELD_MASTERY_TP));
                    }
                }
                else if (PDefender->objtype == TYPE_TRUST && blockRateMod > 0)
                {
                    absorb = 50;

                    // Shield Mastery
                    if ((std::max(damage - (PDefender->getMod(Mod::PHALANX) + PDefender->getMod(Mod::STONESKIN)), 0) > 0) &&
                        (PDefender->getMod(Mod::SHIELD_MASTERY_TP)))
                    {
                        // If the player blocked with a shield and has shield mastery, add shield mastery TP bonus
                        // unblocked damage (before block but as if affected by stoneskin/phalanx) must be greater than zero
                        PDefender->addTP(PDefender->getMod(Mod::SHIELD_MASTERY_TP));
                    }
                }
                else
                {
                    absorb = 50;
                }

                // Reprisal
                if (damage > 0 && PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_REPRISAL))
                {
                    // Reflect a portion of the blocked damage back. This is calculated before Stoneskin, Phalanx, Sentinel or Invincible
                    CStatusEffect* reprisalEffect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL);

                    if (reprisalEffect != nullptr)
                    {
                        float spikesBonus   = 1.f + (PDefender->getMod(Mod::REPRISAL_SPIKES_BONUS) / 100.f);
                        int16 effectPower   = (int16)(reprisalEffect->GetPower() * spikesBonus);
                        int32 blockedDamage = (damage * (100 - absorb)) / 100;
                        int32 spikesDamage  = 0;

                        if (PDefender->StatusEffectContainer->HasStatusEffect({ EFFECT_INVINCIBLE, EFFECT_SENTINEL }))
                        {
                            blockedDamage = (baseDamage * (100.f - absorb)) / 100.f;
                        }

                        spikesDamage = blockedDamage * (effectPower / 100.f);

                        // Set Reprisal spike damage
                        PDefender->setModifier(Mod::SPIKES_DMG, spikesDamage);
                    }
                }
                damage = (damage * absorb) / 100;
            }
```
for get block rate. only falls through block calculations if the mod is on.

```
else if (PDefender->objtype == TYPE_TRUST && PDefender->getMod(Mod::SHIELDBLOCKRATE) > 0)
        {
            base = PDefender->getMod(Mod::SHIELDBLOCKRATE);
        }
```

for reprisal effect, only applies reprisal spikes if mod is present

```
else if (PDefender->objtype == TYPE_TRUST)
        {
             if (PDefender->getMod(Mod::SHIELDBLOCKRATE) > 0)
             {
                  Action->spikesEffect = SUBEFFECT::SUBEFFECT_REPRISAL;
             }
```

Reply:
New changes apply flat block rate controlled by mob:setMod(xi.mod.SHIELDBLOCKRATE, num)

If no mod, no block
If no mod, no reprisal

![image](https://user-images.githubusercontent.com/99501957/198103386-0ea52c0a-d2ef-4de5-9b87-64f687b95d3d.jpeg)

Reply:
I think we should not re-use the block rate mod for this and add 2 new ones:

a mod to indicate that the entity has a shield
a mod to indicate the base block rate when the above mod is non-zero

This way we dont have to touch it again in the future if we figure out that certain trusts have different base block rates

The block rate mod is used to indicate how much more block rate you should get from gear/buffs, and it modifies block rate in a way that is non-linear

Reply:
So we need to create a whole new mobmod that toggles true/false for having a shield. 

Would we want to set block rate using the modifier value as well like I have here so that base rates can be manually adjusted in the Lua in the event they are variable by trust?

Reply:
I nominate backfilling currently unused mobmod ID 19 (since the mod won't even be on players, it makes sense as a mobmod rather than a regular mod_

Reply:
Making sure I have correct thinking,

Call it something like phantomshield then explain it in comments?

Mobmodifier.h - add 19, phantomshield
Status.lua global- add 19, phantomshield

xi.mobmod.PHANTOMSHIELD

Then swap out SHIELDBLOCKRATE with that modifier in Battleutils.cpp?

Reply:
no to "phantom" anything. `MOBMOD_CAN_SHIELD_BLOCK` works

Reply:
Lol ok sorry XD
Will adjust.

Reply:
a separate regular modifier would determine base black rate.

@WinterSolstice8 I don't think he has to add 2 mods for this, just the one for the "it has a shield/can block" part. The existing one is only used by pup automatons currently and appears to simply be its base blockrate since there is no shield "size" available.

I think we want that mobmod tho just to make it clear whats happening

Reply:
Sounds good to me if that's the case, I was probably conflating it with block rate +% gear base on its name alone

Reply:
So we would still use the SHIELDBLOCKRATE to set the base rate via lua and the mobmod to toggle shield exists or not?

Reply:
> Sounds good to me if that's the case, I was probably conflating it with block rate +% gear base on its name alone

it is on some gear, however that fear can't affact non-players at all. so the automaton is using it to give it a non-zero block rate. in the existing code it's getting 45 of that mod when it valoredge form. its not a percent on top of base - it is the base percent rate blocking happens for its nonexistent "shield" (and why bart keeps wanting to say phantom xD).